taichi/system/memory_pool.h

```diff
@@ -2,7 +2,7 @@
 #include "taichi/common/core.h"
 #include "taichi/system/unified_allocator.h"
 #define TI_RUNTIME_HOST
-#include "taichi/runtime/llvm/context.h"
+#include "taichi/runtime/llvm/mem_request.h"
```

Comment:
Forgot to undef?

Reply:
```suggestion
#include "taichi/runtime/llvm/mem_request.h"
#undef TI_RUNTIME_HOST
```

Sounds good. It doesn't harm not to, but maybe an `undef` here makes the code slightly more structured. (I was the original author of this part, so this is actually offtopic.)

Reply:
Hmm, `#undef` does improve the quality. In fact, I've found that some files accidentally depend on `program.h` having defined `TI_RUNTIME_HOST`. But we can clean it up later..

https://github.com/taichi-dev/taichi/blob/7fa0d9dbf689e7059cfdf0fa71db9ffd8279846e/taichi/program/program.h#L9-L20

Reply:
Hmm, it turned out that we cannot add `#undef`  yet, precisely because of the reason mentioned above... Let's systematically fix that in another PR.